File: pipeline/label_structure/structure/prompt.py
STRUCTURAL_METADATA_PROMPT = """Extract structural metadata from three OCR outputs of the same book page.

<mistral_ocr>
{mistral_text}
</mistral_ocr>

<olm_ocr>
{olm_text}
</olm_ocr>

<paddle_ocr>
{paddle_text}
</paddle_ocr>

<headings>
{headings_json}
</headings>

<pattern_hints>
{pattern_hints_json}
</pattern_hints>

**Pattern Hints:**
- `has_mistral_footnote_refs`: Page has [^N] footnote markers
- `mistral_footnote_count`: Number of footnote markers detected
- `has_repeated_symbols`: Multiple instances of *, †, ‡ (footnote indicators)
- `has_mistral_images`: Page contains ![alt](img.jpeg) image references

**Important:** If pattern hints indicate footnotes, be careful not to confuse footnote content at bottom with running footers.

## OCR Quality Check

If ANY OCR output is corrupted (repetitive garbage like "I I I I...", random characters, clearly nonsensical):
- Set ALL observations to `present: false` with `confidence: "low"`
- Use empty strings for text fields
- Skip analysis - don't try to extract from garbage

## Structural Metadata Extraction

**1. Headers (running headers in top margin)**
- Compare Paddle vs Mistral/OLM
- First lines in Paddle MISSING from Mistral/OLM = header
- Typically: book title, chapter name, section name
- If top text matches a heading from Pass 1, it's likely a chapter title, not a running header
- If Paddle ≈ Mistral/OLM at top, no header

**2. Footers (running footers in bottom margin)**
- Compare Paddle vs Mistral/OLM
- Last lines in Paddle MISSING from Mistral/OLM = footer
- Typically: book title, chapter name, author name
- NOT footnote content (check pattern hints for footnote indicators)
- If Paddle ≈ Mistral/OLM at bottom, no footer

**3. Page Numbers**
- Look ONLY in header/footer text identified above
- Patterns: "Page 34", "- 34 -", "34", "xiv" (Roman numerals)
- Location: header, footer, or margin
- NOT body text references: "see page 42", "pages 15-20" are references, not page numbers
- Extract the number/numeral value

**Note:** This is ONLY for structural elements (headers/footers/page_num).
Do NOT confuse footnote content with running footers.

**Confidence:** high (clear) | medium (somewhat clear) | low (uncertain)"""

File: internal/pipeline/prompts/label/prompt.go
package label

// SystemPrompt is the system prompt for label structure extraction.
// Copied from Python: pipeline/label_structure/unified/prompt.py
const SystemPrompt = `You extract page numbers and running headers from book pages.

## Page Numbers

Location: Usually in header or footer region (first/last lines)
Formats: "34", "- 34 -", "Page 34", "xiv" (roman numerals for front matter)
NOT page numbers: "see page 42", "on page 10" (these are references in body text)

## Running Headers

Running headers repeat on consecutive pages within a chapter/section.
Common patterns:
- Chapter title on left pages, book title on right pages
- "Chapter 3: The Beginning" or just "The Beginning"
- Author name on one side, title on other
- Section names in academic texts

Running headers are NOT:
- Chapter headings (# Heading syntax) - these start new sections
- Footnote content at bottom of page
- Body text

Key insight: The same running header text appears on many consecutive pages.`

// UserPromptTemplate is the user prompt template for label extraction.
// Use fmt.Sprintf(UserPromptTemplate, blendedText)
const UserPromptTemplate = `Extract the page number and running header from this page:

%s`
